src/workers/chunkWorker.js

```diff
@@ -0,0 +1,321 @@
+      normals.push(1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0);
+      uvs.push(...onefaceuv);
+      dbstr = dbstr + "left\n";
+      faceIndexMap[facemapcount] = makefaceIndexMapObject(cen, nx - 1, ny, nz);
+      facemapcount += 2;
+    }
+    // //right
+    if (!blocks[makeKey(nx + t2, ny, nz)]) {
+      vertices.push(...c[5], ...c[8], ...c[1]);
+      vertices.push(...c[1], ...c[8], ...c[4]);
+      normals.push(1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0);
+      uvs.push(...onefaceuv);
+      dbstr = dbstr + "right\n";
+      faceIndexMap[facemapcount] = makefaceIndexMapObject(cen, nx + 1, ny, nz);
+      facemapcount += 2;
+    }
+    // //top
+    if (!blocks[makeKey(nx, ny + t2, nz)]) {
+      vertices.push(...c[4], ...c[8], ...c[3]);
+      vertices.push(...c[3], ...c[8], ...c[7]);
+      normals.push(0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0);
+      uvs.push(...onefaceuv);
+      dbstr = dbstr + "top\n";
+      faceIndexMap[facemapcount] = makefaceIndexMapObject(cen, nx, ny + 1, nz);
+      facemapcount += 2;
+    }
+    // //bot
+    if (!blocks[makeKey(nx, ny - t2, nz)]) {
+      vertices.push(...c[5], ...c[1], ...c[6]);
+      vertices.push(...c[6], ...c[1], ...c[2]);
+      normals.push(0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0);
+      uvs.push(...onefaceuv);
+      dbstr = dbstr + "bot\n";
+      faceIndexMap[facemapcount] = makefaceIndexMapObject(cen, nx, ny - 1, nz);
+      facemapcount += 2;
+    }
+
+    blocks[cen].showface = showfaces;
+  });
+
+  return [vertices, uvs, normals, faceIndexMap];
+}
+
+function buildChunkForTheFirstTime(worldCubeSize, chunkNumber) {
+  let ws = worldCubeSize;
+  let cnX = Math.floor(chunkNumber / ws);
+  let cnZ = (chunkNumber / worldCubeSize - cnX) * ws;
+
+  // const prng = alea("1000");
+  const noise2D = worldSet["genNoise2D"];
+  let info = {};
+  let infoList = [];
+  // let xs = worldCubeSize**2;
+  let xs = ws; // world chunk size
+  let ys = 1;
+  let zs = xs;
+  let heightFactor = worldSet["heightFactor"];
+  let depth = 0;
+  let key = "";
+
+  let ty = 0; //test y for noise
+
+  let difflimit = AMTmapkeys.length;
+
+  for (let x = 16 * cnX; x < 16 * cnX + 16; x++) {
+    for (let y = -1 * Math.abs(depth); y < ys; y++) {
+      for (let z = 16 * cnZ; z < 16 * cnZ + 16; z++) {
+        ty = worldSet.showFlatWorld ? y : Math.floor(((noise2D(x / 100, z / 100) + 1) * heightFactor) / 2) + y;
+
+        key = makeKey(x, ty, z);
+        let texture = "";
+        if (worldSet["useHeightTextures"]) {
+          texture = AMTmapkeys[ty % difflimit];
+        } else {
+          texture = Math.abs(x - z) < 16 ? "wood" : "grass";
+        }
+        infoList.push(key);
+        info[key] = {
+          pos: [x, ty, z],
+          texture,
+        };
+      }
+    }
+  }
+  // console.log('myinfo',info)
+
+  // REF_ALLCUBES.current = { ...REF_ALLCUBES.current, ...info }; // i need to add the new blocks to all cubes ref
```

Comment:
i think this todo comment was already fixed? If so, lets get rid of it. Not fully sure though tbh